Lib/sysconfig/__init__.py

```diff
@@ -414,6 +414,13 @@ def _init_non_posix(vars):
     vars['BINDIR'] = os.path.dirname(_safe_realpath(sys.executable))
     vars['TZPATH'] = ''
 
+    abiflags = vars['abi_thread']
+    if vars['EXT_SUFFIX'].endswith('_d.'):
+        abiflags += 'd'
+    # Do not touch lower-cased `abiflags` here.
+    # sys.abiflags is absent on Windows. So vars['abiflags'] should be empty.
+    vars['ABIFLAGS'] = abiflags
```

Comment:
Maybe we can move this to `_sysconfig.config_vars()` in C code.

https://github.com/python/cpython/blob/972a295fe34280aa3d16c573d6200025a1ce4ff0/Lib/sysconfig/__init__.py#L404-L405